libraries/AP_Arming/AP_Arming.cpp

```diff
@@ -369,6 +369,26 @@ bool AP_Arming::gps_checks(bool report)
             return false;
         }
     }
+
+    // check GPSs are within 50m of each other and that blending is healthy
+    if ((checks_to_perform & ARMING_CHECK_ALL) || (checks_to_perform & ARMING_CHECK_GPS_CONFIG)) {
+        float distance_m;
+        if (!gps.all_consistent(distance_m)) {
+            if (report) {
+                GCS_MAVLINK::send_statustext_all(MAV_SEVERITY_CRITICAL,
+                                                 "PreArm: GPSs inconsistent by %4.1fm",
+                                                  (double)distance_m);
+            }
+            return false;
+        }
+        if (!gps.blend_healthy()) {
```

Comment:
I guess there's two points here.
First one is the naming of the pre-arm check.  I don't think splitting it into two methods, "blend_active()" and "blend_healthy()" is really adding any value.  The check is only used in this one place.  We could perhaps rename it to "blend_health_check".
Second is about the race condition on startup.  Remember to take-off in a mode requiring GPS, the EKF must also be happy about it's position.  That takes at least a few seconds after the GPS updates start arriving.  So in practice I don't think it can happen.  The solution to 100% guard against what you're suggesting is to add another counter that ensure we've had at least X seconds of data.. but it seems to me the additional complexity for a problem that can't happen in practice probably isn't worth it.